source/distributed/fully_distributed_tria.cc

```diff
@@ -0,0 +1,519 @@
+    bool
+    Triangulation<dim, spacedim>::has_hanging_nodes() const
+    {
+      AssertThrow(false, ExcNotImplemented());
+      return false;
+    }
+
+
+
+    template <int dim, int spacedim>
+    std::size_t
+    Triangulation<dim, spacedim>::memory_consumption() const
+    {
+      std::size_t mem =
+        this->dealii::parallel::TriangulationBase<dim, spacedim>::
+          memory_consumption() +
+        MemoryConsumption::memory_consumption(
+          coarse_cell_id_to_coarse_cell_index_vector) +
+        MemoryConsumption::memory_consumption(
+          coarse_cell_index_to_coarse_cell_id_vector);
+      return mem;
+    }
+
+
+
+    template <int dim, int spacedim>
+    std::map<unsigned int, std::set<dealii::types::subdomain_id>>
+    Triangulation<dim, spacedim>::compute_vertices_with_ghost_neighbors() const
+    {
+      std::vector<bool> vertex_of_own_cell(this->n_vertices(), false);
+
+      // collect nodes coinciding due to periodicity
+      std::map<unsigned int, unsigned int> periodic_map;
+      for (auto &cell : this->active_cell_iterators())
+        if (cell->is_locally_owned() || cell->is_ghost())
+          {
+            for (unsigned int i = 0; i < GeometryInfo<dim>::faces_per_cell; i++)
+              {
+                if (cell->has_periodic_neighbor(i) &&
+                    cell->periodic_neighbor(i)->active())
+                  {
+                    auto face_t = cell->face(i);
+                    auto face_n = cell->periodic_neighbor(i)->face(
+                      cell->periodic_neighbor_face_no(i));
+                    for (unsigned int j = 0;
+                         j < GeometryInfo<dim>::vertices_per_face;
+                         j++)
+                      {
+                        auto         v_t  = face_t->vertex_index(j);
+                        auto         v_n  = face_n->vertex_index(j);
+                        unsigned int temp = std::min(v_t, v_n);
+                        {
+                          auto it = periodic_map.find(v_t);
+                          if (it != periodic_map.end())
+                            temp = std::min(temp, it->second);
+                        }
+                        {
+                          auto it = periodic_map.find(v_n);
+                          if (it != periodic_map.end())
+                            temp = std::min(temp, it->second);
+                        }
+                        periodic_map[v_t] = temp;
+                        periodic_map[v_n] = temp;
+                      }
+                  }
+              }
+          }
+
+      // compress map
+      for (auto &p : periodic_map)
+        {
+          if (p.first == p.second)
+            continue;
+          unsigned int temp = p.second;
+          while (temp != periodic_map[temp])
+            temp = periodic_map[temp];
+          p.second = temp;
+        }
+
+      // check if map is actually compressed
+      for (auto p : periodic_map)
```

Comment:
Is this something we should only do in debug mode? If yes, surround by ``#ifdef DEBUG``.